Libraries/src/Amazon.Lambda.Annotations.SourceGenerator/Templates/LambdaFunctionTemplate.tt

```diff
@@ -227,6 +237,21 @@ namespace <#= _model.LambdaMethod.ContainingNamespace #>
 <#
                 }
             }
+            else if (parameter.Attributes.Any(att => att.Type.FullName == TypeFullNames.FromRouteAttribute) || routeParameters.Contains(parameter.Name))
+            {
+                var fromRouteAttribute = parameter.Attributes?.FirstOrDefault(att => att.Type.FullName == TypeFullNames.FromRouteAttribute) as AttributeModel<FromRouteAttribute>;
+
+                // Use parameter name as key, if Name has not specified explicitly in the attribute definition.
+                var routeKey = fromRouteAttribute?.Data?.Name ?? parameter.Name;
+#>
+            var <#= parameter.Name #> = default(<#= parameter.Type.FullName #>);
+            if (request.PathParameters?.ContainsKey("<#= routeKey #>") == true)
+            {
+                <#= parameter.Name #> = (<#= parameter.Type.FullName #>)Convert.ChangeType(request.PathParameters["<#= routeKey #>"], typeof(<#= parameter.Type.FullName #>));
```

Comment:
We also need to think about all of these `ChangeType` calls how to better handle if the type can't be converted. This is user input data which can be wrong. Should the end result we automatically send back an 400 bad request saying this parameter is the wrong data type?

Reply:
I think that's fair, a bad request can be sent on type conversion failure. I will check in ASP.NET Core as well to make sure the behavior is same/similar.